views/js/tools/diagnostic/diagnostic.js

```diff
@@ -293,97 +293,122 @@ define([
-                            self.addResult(status);
-                            cb();
-                        });
+                if (self.is('rendered')) {
+                    // set up the component to a new run
+                    self.prepare();
+
+                    _.forEach(self.config.testers, function(testerConfig, testerId) {
+                        testerConfig.id = testerConfig.id || testerId;
+                        if (testerConfig.enabled) {
+                            testers.push(function (cb) {
+                                doCheck(testerConfig, cb);
+                            });
+                        }
                     });
-                });
-            }
 
-            if (this.is('rendered')) {
-                // set up the component to a new run
-                this.prepare();
-
-                _.forEach(this.config.testers, function(testerConfig, testerId) {
-                    testerConfig.id = testerConfig.id || testerId;
-                    if (testerConfig.enabled) {
-                        testers.push(function (cb) {
-                            doCheck(testerConfig, cb);
-                        });
-                    }
-                });
+                    // launch each testers in series, then display the results
+                    async.series(testers, function () {
+                        // pick the lowest percentage as the main score
+                        var total = _.min(scores, 'percentage');
 
-                // launch each testers in series, then display the results
-                async.series(testers, function () {
-                    // pick the lowest percentage as the main score
-                    var total = _.min(scores, 'percentage');
+                        // get a status according to the main score
+                        var status = getStatus(total.percentage, _thresholds);
 
-                    // get a status according to the main score
-                    var status = getStatus(total.percentage, _thresholds);
+                        // display the result
+                        status.title = __('Total');
+                        status.id = 'total';
+                        self.addCustomFeedbackMsg(status, self.config.configurableText.diagTotalCheckResult);
 
-                    // display the result
-                    status.title = __('Total');
-                    status.id = 'total';
-                    self.addCustomFeedbackMsg(status, self.config.configurableText.diagTotalCheckResult);
+                        status.details = information;
+                        self.addResult(status);
 
-                    status.details = information;
-                    self.addResult(status);
+                        // done !
+                        self.finish();
+                    });
+                }
 
-                    // done !
-                    self.finish();
-                });
+                return self;
```

Comment:
This line is not needed here. It should be put back at the previous place, this is the `run()` method that needs the chain return.